.github/workflows/functional-test.yaml

```diff
@@ -177,6 +177,12 @@ jobs:
         with:
           python-version: 3.9
 
+      - name: Install python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install numpy
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

Comment:
We don't have a `requirements.txt` at the moment, but I figured this would be nice to keep in case we decide to add one. Good to note that if we want to use pandas for the analysis script, we can just add it to the `pip install` line above this as a simple alternative